crates/driver/src/boundary/liquidity/mod.rs

```diff
@@ -0,0 +1,95 @@
+use {
+    crate::{
+        boundary,
+        domain::{competition::auction, eth, liquidity},
+        infra::blockchain::Ethereum,
+    },
+    anyhow::Result,
+    futures::future,
+    itertools::Itertools,
+    shared::{
+        baseline_solver::BaseTokens,
+        current_block::{self, BlockRetrieverStrategy, CurrentBlockStream},
+        recent_block_cache::CacheConfig,
+    },
+    solver::liquidity_collector::LiquidityCollector,
+    std::{
+        num::{NonZeroU64, NonZeroUsize},
+        sync::Arc,
+        time::Duration,
+    },
+};
+
+mod uniswap;
+
+/// The default poll interval for the block stream updating task.
+const BLOCK_POLL_INTERVAL: Duration = Duration::from_secs(1);
+
+/// The default pool caching configuration to use.
+fn cache_config() -> CacheConfig {
+    CacheConfig {
+        number_of_blocks_to_cache: NonZeroU64::new(10).unwrap(),
+        number_of_entries_to_auto_update: NonZeroUsize::new(200).unwrap(),
+        maximum_recent_block_age: 4,
+        max_retries: 5,
+        delay_between_retries: Duration::from_secs(1),
+    }
+}
+
+pub struct Fetcher {
+    blocks: CurrentBlockStream,
+    inner: LiquidityCollector,
+}
+
+impl Fetcher {
+    /// Creates a new fether for the specified configuration.
+    pub async fn new(eth: &Ethereum, config: &liquidity::fetcher::Config) -> Result<Self> {
+        let blocks = current_block::Arguments {
+            block_stream_poll_interval_seconds: BLOCK_POLL_INTERVAL,
+            block_stream_retriever_strategy: BlockRetrieverStrategy::EthCall,
+        }
+        .stream(boundary::web3(eth))
+        .await?;
```

Comment:
Isn't the block stream more of a global component that should be passed in?

Reply:
I considered this, but the only place this is currently used is here. If we need to create a block stream for something else, then I agree that we should pass it in.

Reply:
This will probably be forgotten if someone else needs a pool cache. I'd rather pass it in but it's no big deal either.